staging/src/k8s.io/client-go/plugin/pkg/client/auth/exec/exec_test.go

```diff
@@ -1262,3 +1269,184 @@ func genClientCert(t *testing.T) ([]byte, []byte) {
 	return pem.EncodeToMemory(&pem.Block{Type: "CERTIFICATE", Bytes: certRaw}),
 		pem.EncodeToMemory(&pem.Block{Type: "PRIVATE KEY", Bytes: keyRaw})
 }
+
+func TestTLSCredentialsCallsAndRotation(t *testing.T) {
+	now := time.Now()
+
+	server := httptest.NewUnstartedServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		fmt.Fprintln(w, "ok")
+	}))
+	server.TLS = &tls.Config{
+		ClientAuth: tls.RequireAnyClientCert,
+	}
+	server.StartTLS()
```

Comment:
server.EnableHTTP2 = true?

Reply:
Not sure why that would matter here?

Reply:
because http1 and http2 behave a lot different, per example, http2 doesn't create new connection if there are existing ones in the pool, you don't have to read close.

And most of the kubernetes environments use http2 these days, I'd say that if we test only one protocol, it should be http2, otherwise we should test both, specially for connection bugs, there are huge differences on how the clients behave

Reply:
but you are right, this is not related to the PR
https://github.com/kubernetes/kubernetes/pull/112404#discussion_r971598512